sdk/tests/deqp/functional/gles3/es3fShaderDerivateTests.js

```diff
@@ -647,7 +649,27 @@ goog.scope(function() {
         /** @type {Array<number>} */ var indices = [0, 2, 1, 2, 3, 1];
 
         gl.clearColor(0.125, 0.25, 0.5, 1.0);
-        gl.clear(gl.COLOR_BUFFER_BIT | gl.DEPTH_BUFFER_BIT | gl.STENCIL_BUFFER_BIT);
+        // We can't really call clear() on gl.COLOR_BUFFER_BIT here as in c++ deqp.
+        // The fbo format might be of integer type and WebGL2 requires an INVALID_OPERATION to be generated.
+        var formatObj = gluTextureUtil.mapGLInternalFormat(fboFormat);
+	var fmtClass = tcuTexture.getTextureChannelClass(formatObj.type);
+	switch (fmtClass) {
+            case tcuTexture.TextureChannelClass.FLOATING_POINT:
+            case tcuTexture.TextureChannelClass.SIGNED_FIXED_POINT:
+            case tcuTexture.TextureChannelClass.UNSIGNED_FIXED_POINT:
+                gl.clear(gl.COLOR_BUFFER_BIT | gl.DEPTH_BUFFER_BIT | gl.STENCIL_BUFFER_BIT);
+                break;
+            case tcuTexture.TextureChannelClass.UNSIGNED_INTEGER:
+                gl.clearBufferuiv(gl.COLOR, 0, new Uint32Array(4));
```

Comment:
The only question is whether the clear color is deliberately non-zero in order to put something that isn't solid black (or transparent black, in this case) into the back buffer.

No concrete suggestion. I might suggest multiplying the clear color above by 255 and putting those values into these Int32Arrays and Uint32Arrays, if the test still passes by doing so. Otherwise, your code as is looks fine.